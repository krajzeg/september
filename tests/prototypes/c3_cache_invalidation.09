# create a deeper inheritance structure

A := Object()
B := Object()
C := Object()
DP1 := [[ deepProperty: 42 ]]
DP2 := [[ deepProperty: 23 ]]
A.prototypes = [B, C]
B.prototypes = DP1

print("In the beginning, this should be 42:", A.deepProperty)

# change prototypes somewhere deep in the inheritance graph
B.prototypes = Object
C.prototypes = [String, DP2]

print("After switching prototypes, it should be 23:", A.deepProperty)

# change prototypes back
B.prototypes = DP1
print("After switching them back, 42 again:", A.deepProperty)
