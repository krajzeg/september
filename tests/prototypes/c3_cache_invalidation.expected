In the beginning, this should be 42: 42
After switching prototypes, it should be 23: 23
After switching them back, 42 again: 42
